utils/economic-checks/src/util.rs

```diff
@@ -0,0 +1,342 @@
+    UncheckedExtrinsic, Usage,
+};
+use parking_lot::RwLock;
+use rand::{rngs::StdRng, RngCore};
+use sp_consensus_aura::{sr25519::AuthorityId as AuraId, Slot, AURA_ENGINE_ID};
+use sp_core::{
+    offchain::{
+        testing::{PoolState, TestOffchainExt, TestTransactionPoolExt},
+        Duration, OffchainDbExt, OffchainWorkerExt, TransactionPoolExt,
+    },
+    sr25519, Pair, Public,
+};
+use sp_finality_grandpa::AuthorityId as GrandpaId;
+use sp_io::offchain;
+use sp_runtime::{
+    traits::{IdentifyAccount, Verify},
+    AccountId32, Digest, DigestItem,
+};
+
+use sp_std::collections::btree_map::BTreeMap;
+use std::sync::Arc;
+
+type GasNodeKeyOf<T> = <<T as pallet_gear::Config>::GasHandler as ValueTree>::Key;
+type GasBalanceOf<T> = <<T as pallet_gear::Config>::GasHandler as ValueTree>::Balance;
+
+pub(crate) type WaitlistOf<T> = <<T as pallet_gear::Config>::Messenger as Messenger>::Waitlist;
+
+// Generate a crypto pair from seed.
+pub(crate) fn get_from_seed<TPublic: Public>(seed: &str) -> <TPublic::Pair as Pair>::Public {
+    TPublic::Pair::from_string(&format!("//{}", seed), None)
+        .expect("static values are valid; qed")
+        .public()
+}
+
+type AccountPublic = <Signature as Verify>::Signer;
+
+// Generate an account ID from seed.
+pub(crate) fn get_account_id_from_seed<TPublic: Public>(seed: &str) -> AccountId32
+where
+    AccountPublic: From<<TPublic::Pair as Pair>::Public>,
+{
+    AccountPublic::from(get_from_seed::<TPublic>(seed)).into_account()
+}
+
+// Generate an Aura authority key.
+pub(crate) fn authority_keys_from_seed(s: &str) -> (AccountId32, AuraId, GrandpaId) {
+    (
+        get_account_id_from_seed::<sr25519::Public>(s),
+        get_from_seed::<AuraId>(s),
+        get_from_seed::<GrandpaId>(s),
+    )
+}
+
+pub(crate) fn create_random_accounts(
+    rng: &mut StdRng,
+    root_acc: &AccountId32,
+) -> Vec<(AccountId32, u128)> {
+    let initial_accounts_num = 1 + (rng.next_u32() % 1000); // [1..1000]
+    let mut accounts = vec![(root_acc.clone(), 1_000_000_000_000_000_u128)];
+    for _ in 1..initial_accounts_num {
+        let mut acc_id = [0_u8; 32];
+        rng.fill_bytes(&mut acc_id);
+        let balance = (rng.next_u64() >> 14) as u128; // approx. up to 10^15
+        accounts.push((acc_id.into(), balance));
+    }
+    accounts
+}
+
+pub(crate) fn new_test_ext(
+    balances: Vec<(impl Into<AccountId32>, u128)>,
+    initial_authorities: Vec<(AccountId32, AuraId, GrandpaId)>,
+    root_key: AccountId32,
+) -> sp_io::TestExternalities {
+    let mut t = system::GenesisConfig::default()
+        .build_storage::<Runtime>()
+        .unwrap();
+
+    pallet_balances::GenesisConfig::<Runtime> {
+        balances: balances
+            .into_iter()
+            .map(|(acc, balance)| (acc.into(), balance))
+            .chain(
+                initial_authorities
+                    .iter()
+                    .cloned()
+                    .map(|(acc, _, _)| (acc, 1000)),
```

Comment:
can't we move that 1000 as the fn argument for future changing? is it enough for any action?